sw/device/lib/dif/dif_mbx.c

```diff
@@ -57,3 +58,56 @@ dif_result_t dif_mbx_range_get(const dif_mbx_t *mbx,
       mmio_region_read32(mbx->base_addr, MBX_OUTBOUND_LIMIT_ADDRESS_REG_OFFSET);
   return kDifOk;
 }
+
+dif_result_t dif_mbx_is_busy(const dif_mbx_t *mbx, bool *is_busy) {
+  if (mbx == NULL || is_busy == NULL) {
+    return kDifBadArg;
+  }
+
+  *is_busy = (bitfield_bit32_read(
+                  mmio_region_read32(mbx->base_addr, MBX_STATUS_REG_OFFSET),
+                  MBX_STATUS_BUSY_BIT) == 1);
+  return kDifOk;
+}
+
+dif_result_t dif_mbx_process_request(const dif_mbx_t *mbx,
```

Comment:
This function is a bit suspicious: it reads data from the mailbox into `request->data_words` but there seems to be no limit to how much data can be read (except for the mailbox limit), and it does not even take as input the maximum number of words that can be written.
I would suggest that this function uses `request->nr_dwords` as the maximum size, and updates it with the number of words written. Possibly it should indicate as well when it has reached the maximum but there was still more to read?

Reply:
Agree. In the last commit I changed the function to limit the number words read by `request->nr_dwords`. If there is more to read, I'm returning a different return code.